framework/include/utils/PiecewiseByBlockLambdaFunctor.h

```diff
@@ -169,7 +169,8 @@ PiecewiseByBlockLambdaFunctor<T>::setFunctor(const MooseMesh & libmesh_dbg_var(m
 
   // Handle special case of ANY_BLOCK_ID and empty block restriction that also cover
   // INVALID_BLOCK_ID
-  if (block_ids.count(Moose::ANY_BLOCK_ID) || block_ids.empty())
+  if (block_ids.count(Moose::ANY_BLOCK_ID) || block_ids.empty() ||
+      block_ids == mesh.meshSubdomains())
     add_lammy(Moose::INVALID_BLOCK_ID);
```

Comment:
Can we change this to `Moose::ANY_BLOCK_ID`? Every time I look at this code, I'm confused

Reply:
no we cant because the goal is to work with this
```
const std::set<SubdomainID> NodeArg::undefined_subdomain_connection = {Moose::INVALID_BLOCK_ID};
```

Reply:
K I have a change I'm going to make a PR for. I think we need a distinction between unknown but valid vs. invalid